app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt

```diff
@@ -0,0 +1,894 @@
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 9)
+      scrollToPosition(position = 10)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 10)
+      scrollToPosition(position = 12)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 12)
+      scrollToPosition(position = 13)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 13)
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_selectChaptersOfFirstStoryInOrder_chaptersAreChecked() {
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 2)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 2)
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_selectChaptersOfFirstStoryInOrder_configChange_chaptersAreChecked() { // ktlint-disable max-line-length
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 2)
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 1)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 2)
+      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 2)
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_firstStory_firstChapterUnchecked_secondChapterIsDisabled() {
+    launch<MarkChaptersCompletedTestActivity>(
+      createMarkChaptersCompletedTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 1,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isChecked())))
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.mark_chapters_completed_recycler_view,
+          position = 2,
+          targetViewId = R.id.mark_chapters_completed_chapter_check_box
+        )
+      ).check(matches(not(isEnabled())))
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedFragment_configChange_firstStory_firstChapterUnchecked_secondChapterIsDisabled() { // ktlint-disable max-line-length
```

Comment:
Shouldn't `configChange` happen after the action has been done? The idea is to verify that the intended functionality doesn't break due to this configuration change. I think here we could do something like -- `testMarkChaptersCompletedFragment_firstStory_firstChapterUnchecked_secondChapterIsDisabled_configChange_secondChapterRemainsDisabled()`
Ditto for everywhere else in this file.

Reply:
I think in this particular test, we are testing this for the landscape configuration as we are not performing any action here which we should check on configChange. So, I'm replacing `configChange` with `land`.

Reply:
I will update other tests as well based on what we are trying to test.

Reply:
Done.

Reply:
Done.